reject-escalated-privileges-yaml/reject-escalated-privileges-yaml.rb

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env ruby
+
+require "json"
+require "octokit"
+require "yaml"
+
+require File.join(File.dirname(__FILE__), "github")
+
+# can expand this list spliting with spaces. e.g %w(cluster-admin root webops)
+STRING_LIST = %w[cluster-admin]
+
+def yaml_files(gh)
+  yaml_files_in_pr(gh).find_all { |file| return has_privileges(file) }
+end
+
+def has_privileges(file)
+  hash = YAML.load_file(file)
+  pattern = Regexp.union(STRING_LIST)
+  recurse(hash, pattern) do |path, value|
+    line = "#{path}:\t#{value}"
+    line = line.gsub(pattern_text) { |match| match }
+    return
+  end
+end
+
+def recurse(obj, pattern, current_path = [], &block)
+  if obj.is_a?(String)
+    path = current_path.join(".")
+    if obj =~ pattern || path =~ pattern
+      yield [path, obj]
+    end
+  elsif obj.is_a?(Hash)
+    obj.each do |key, value|
+      recurse(value, pattern, current_path + [key], &block)
+    end
+  elsif obj.is_a?(Array)
+    obj.each do |value|
+      recurse(value, pattern, current_path, &block)
+    end
+  end
+end
+
+# Attempt to parse all the yaml/yml files in a PR,
+# aside from those with 'secret' in the filename.
+# Files with 'secret' in the name are very often
+# git-crypted, and so would cause this action to
+# fail.
+def yaml_files_in_pr(gh)
+  gh.files_in_pr
+    .grep(/\.(yaml|yml)$/)
+    .reject { |f| f =~ /secret/ }
+end
+
+############################################################
+
+gh = GithubClient.new
+
+privileges_code = yaml_files(gh)
+
+unless privileges_code
```

Comment:
I'm not quite clear how this works "privileges_code" (which might be clearer to name "privileged_yaml_files") is a list of "bad" files, right?

unless == "if not", so it seems as if this should report an error if there aren't any such yaml files? (this is one of the reasons I try to avoid unless - it's confusing)

Should it be:
```
if privileges_code.any?
```
Also, if you've got the names of the 'bad' files at this point, it might be helpful for the action to log them, so the user knows what to fix.